packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1286,3 +1292,39 @@ DevelopmentArtifact _artifactFromTargetPlatform(TargetPlatform targetPlatform) {
 
 /// Returns true if s is either null, empty or is solely made of whitespace characters (as defined by String.trim).
 bool _isBlank(String s) => s == null || s.trim().isEmpty;
+
+
+/// Updates dartdefines based on [webRenderer].
+List<String> _updateDartDefines(List<String> original, String webRenderer) {
+  final List<String> dartDefines = List<String>.from(original);
+  final Iterable<String> webRendererDartDefines = _dartDefinesForWebRenderer(webRenderer);
+  if (webRendererDartDefines.isNotEmpty) {
+    // If --web-renderer is used, removes existing dartDefine that starts with
+    // "FLUTTER_WEB_USE_SKIA="
+    dartDefines.removeWhere((String d) => d.startsWith('FLUTTER_WEB_USE_SKIA='));
+    dartDefines.addAll(webRendererDartDefines);
+  }
+  return dartDefines;
+}
+
+Iterable<String> _dartDefinesForWebRenderer(String webRenderer) {
+  if (!_webRendererDartDefines.containsKey(webRenderer)) {
```

Comment:
This is unnecessarily defensive, since the string arg has allowed values you can trust that only those will be provided